crypto/nrf_cc310_platform/src/nrf_cc3xx_platform_mutex_zephyr.c

```diff
@@ -227,6 +228,47 @@ static int32_t mutex_unlock_platform(nrf_cc3xx_platform_mutex_t *mutex) {
     return NRF_CC3XX_PLATFORM_SUCCESS;
 }
 
+#elif CONFIG_CC3XX_ATOMIC_LOCK
+
+atomic_t sym_mutex;
+atomic_t asym_mutex;
+atomic_t rng_mutex;
```

Comment:
The rng_mutex is very difficult to control since it can be called by anything in Zephyr that needs entropy. This merits handing RNG in a special way. Please change rng_mutex back to use a kernel mutex and handle the additional type difference with a new flag in nrf_cc3xx_platform_mutex.h

_The new flag could be called NRF_CC3XX_PLATFORM_MUTEX_MASK_IS_ATOMIC, so there is a way to distinguish between regular kernel mutex and atomic mutex_